src/test/unit/common/languageServer/languageServer.test.ts

```diff
@@ -144,17 +108,76 @@ suite('Language Server', () => {
 
     languageServer = new LanguageServer(
       user,
-      configuration,
+      configurationMock,
       lca as unknown as ILanguageClientAdapter,
       stubWorkspaceConfiguration('http.proxy', expectedProxy),
       windowMock,
-      authService,
+      authServiceMock,
       new LoggerMock(),
-      downloadService,
+      downloadServiceMock,
     );
-    downloadService.downloadReady$.next();
+    downloadServiceMock.downloadReady$.next();
     await languageServer.start();
     sinon.assert.called(lca.create);
     sinon.verify();
   });
+
+  suite('LanguageServer is initialized', () => {
+    setup(() => {
+      const mockLanguageClient = {
+        start: sinon.stub().resolves(),
+      };
+      const mockLanguageClientAdapter = {
+        create: sinon.stub().returns(mockLanguageClient),
+        getLanguageClient: sinon.stub().returns(mockLanguageClient),
+      };
+      languageServer = new LanguageServer(
+        user,
+        configurationMock,
+        mockLanguageClientAdapter,
+        {} as IVSCodeWorkspace,
+        windowMock,
+        authServiceMock,
+        new LoggerMock(),
+        downloadServiceMock,
+      );
+    });
+
+    test('LanguageServer should provide correct initialization options', async () => {
+      const expectedInitializationOptions: InitializationOptions = {
+        activateSnykCodeSecurity: 'false',
+        activateSnykCodeQuality: 'false',
+        activateSnykOpenSource: 'false',
+        activateSnykIac: 'true',
+        token: 'testToken',
+        cliPath: 'testPath',
+        enableTelemetry: 'true',
+        sendErrorReports: 'true',
+        integrationName: 'VS_CODE',
+        integrationVersion: '0.0.0',
+        automaticAuthentication: 'false',
+        endpoint: undefined,
+        organization: undefined,
+        additionalParams: '--all-projects',
+        manageBinariesAutomatically: 'true',
+        deviceId: user.anonymousId,
+        filterSeverity: { critical: true, high: true, medium: true, low: true },
+        enableTrustedFoldersFeature: 'true',
+        trustedFolders: ['/trusted/test/folder'],
+        insecure: 'true',
+        scanningMode: 'auto',
+      };
+
+      deepStrictEqual(await languageServer.getInitializationOptions(), expectedInitializationOptions);
+    });
+
+    ['auto', 'manual'].forEach(expectedScanningMode => {
```

Comment:
Do we really need to test the `string` setter with different values, when compiler provides this verification? What scenario are we trying to test here?

Reply:
We're testing whether the LS initialization options respond to changes in the configuration, we're not testing the same property

Reply:
The above test is covering this already tho.

Reply:
It only checks the default values, it doesn't cover the manual case